index.js

```diff
@@ -54,3 +61,14 @@ const endWhenStreamsDone = async (passThroughStream, streams) => {
 		}
 	}
 };
+
+const onInputStreamsEnd = async (streams, passThroughStream, {signal}) => {
+	await Promise.all(streams.map(stream => finished(stream, {signal, cleanup: true, readable: true, writable: false})));
+	passThroughStream.end();
+};
+
+const onMergedStreamEnd = async (passThroughStream, {signal}) => {
+	try {
+		await finished(passThroughStream, {signal, cleanup: true});
+	} catch {}
```

Comment:
The purpose of this function is only to resolve when `passThroughStream` ends, aborts or errors. When it does, `Promise.race()` resolves, which calls `abortController.abort()`, which cleans up the other `finished()` calls.